server/src/routers/coordinatesRouter.js

```diff
@@ -0,0 +1,46 @@
+const express = require('express')
+const router = new express.Router()
+
+const Coordinates = require('../models/coordinates')
+
+const coordinatesValidator = require('../middlewares/coordinatesHandlers')
+
+//The distance should be in meters from which we need to find the parking ads
+router.get('/search', coordinatesValidator, async (req, res, next) => {
+    const longitude = req.body.longitude
+    const latitude = req.body.latitude
+
+    try {
+        let parkingAds = await Coordinates.find({
+                location: {
+                    $nearSphere: {
+                        $geometry: {
+                            type: "Point",
+                            coordinates: [longitude, latitude],
+                        },
+                        $minDistance: 0,
+                        $maxDistance: req.body.distance
+                    }
+                }
+            })
+            .populate('parkingId')
+
+        parkingAds = parkingAds.map((ad) => {
+            return {
+                coordinates: ad.location.coordinates,
+                parkingAd: ad.parkingId
```

Comment:
Why do we just return the coordinates and parkingId here? Don't we need other parking information such as title?

Reply:
All the information about a parking ad is stored in parking ad only. Is that what u r asking? If yes, then you will receive everything within this. However, I just noticed that we are not even storing any titles for a parking ad.